Improve LayoutLM (#9476)
huggingface/transformers: docs/source/model_doc/layoutlm.rst

  400,000 images belonging to one of 16 classes).

The abstract from the paper is the following:

*Pre-training techniques have been verified successfully in a variety of NLP tasks in recent years. Despite the
widespread use of pretraining models for NLP applications, they almost exclusively focus on text-level manipulation,
while neglecting layout and style information that is vital for document image understanding. In this paper, we propose
the LayoutLM to jointly model interactions between text and layout information across scanned document images, which is
beneficial for a great number of real-world document image understanding tasks such as information extraction from
scanned documents. Furthermore, we also leverage image features to incorporate words' visual information into LayoutLM.
To the best of our knowledge, this is the first time that text and layout are jointly learned in a single framework for
document-level pretraining. It achieves new state-of-the-art results in several downstream tasks, including form
understanding (from 70.72 to 79.27), receipt understanding (from 94.02 to 95.24) and document image classification
(from 93.07 to 94.42).*

Tips:

> - LayoutLM has an extra input compared to BERT called :obj:`bbox`, which are the bounding boxes (i.e. 2D-positions) of

Reviewer: ```suggestion
- In addition to `input_ids`, :meth:`~transformer.LayoutLMModel.forward` also expects the input :obj:`bbox`, which are the bounding boxes (i.e. 2D-positions) of
```

  the input tokens. These can be obtained using an external OCR engine such as Google's `Tesseract
  <https://pypi.org/project/pytesseract/>`__. Each bounding box should be in (x0, y0, x1, y1) format, where (x0, y0)

>   corresponds to the position of the upper left in the bounding box, and (x1, y1) represents the position of the lower
>   right. Note that one first needs to normalize the bounding boxes to be on a 0-1000 scale. To normalize, you can use

Reviewer 2: ```suggestion
  corresponds to the position of the upper left corner in the bounding box, and (x1, y1) represents the position of the lower
  right corner. Note that one first needs to normalize the bounding boxes to be on a 0-1000 scale. To normalize, you can use
```

  the following function:

.. code-block::

   def normalize_bbox(bbox, width, height):
        return [
            int(1000 * (bbox[0] / width)),
            int(1000 * (bbox[1] / height)),
            int(1000 * (bbox[2] / width)),
            int(1000 * (bbox[3] / height)),
        ]

Here, :obj:`width` and :obj:`height` correspond to the width and height of the original document in which the token
occurs. Those can be obtained using the Python Image Library (PIL) library for example, as follows:

.. code-block::

   from PIL import Image

   image = Image.open("name_of_your_document - can be a png file, pdf, etc.")

>    width, height = image.size

Reviewer: that's very nice
